### PR TITLE
[FLINK-20247][config] Hide unused unaligned checkpoint configuration options

### PR DESCRIPTION
**Pull request by AHeise**
## What is the purpose of the change

As long as the related unaligned checkpoint features rescaling and timeout are not finished, users shouldn't see the related config options. Defaults should be adjusted to show the behavior of unaligned checkpoint without these features.

## Brief change log

- Hide timeout option.
- Hide force unaligned checkpoint option.

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (**not applicable** / docs / JavaDocs / not documented)

### CODE REVIEW

File: flink-streaming-java/src/main/java/org/apache/flink/streaming/api/environment/ExecutionCheckpointingOptions.java [hunk 1]
```diff
@@ -162,10 +162,11 @@
 					"will timeout and checkpoint barrier will start working as unaligned checkpoint.")
 				.build());
 
+	@Documentation.ExcludeFromDocumentation("Do not advertise this option until rescaling of unaligned checkpoint is completed.")
 	public static final ConfigOption<Boolean> FORCE_UNALIGNED =
 		ConfigOptions.key("execution.checkpointing.unaligned.forced")
 			.booleanType()
-			.defaultValue(false)
+			.defaultValue(true)
```
Comment (rkhachatryan) [inline flink-streaming-java/src/main/java/org/apache/flink/streaming/api/environment/ExecutionCheckpointingOptions.java:169]: Why is this change needed?

File: flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/StreamTask.java [hunk 2]
```diff
@@ -501,7 +501,7 @@ protected void beforeInvoke() throws Exception {
 		actionExecutor.runThrowing(() -> {
 
 			SequentialChannelStateReader reader = getEnvironment().getTaskStateManager().getSequentialChannelStateReader();
-			reader.readOutputData(getEnvironment().getAllWriters(), !configuration.isGraphContainingLoops());
+			reader.readOutputData(getEnvironment().getAllWriters(), false);
```
Comment (rkhachatryan) [inline flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/StreamTask.java:504]: nit: add todo and mention the reason?

File: flink-streaming-java/src/main/java/org/apache/flink/streaming/api/environment/ExecutionCheckpointingOptions.java (hunk 1 above)
Comment (AHeise) [reply flink-streaming-java/src/main/java/org/apache/flink/streaming/api/environment/ExecutionCheckpointingOptions.java:169]: Without this change UC wouldn't work with iterations. Now that there is no `EndOfChannelStateEvent` and with it no blocking, we can also support iterations for the time being.
Alternatively, we could keep the force option and discourage using UC in iterations already in 1.12.

Comment (rkhachatryan) [reply flink-streaming-java/src/main/java/org/apache/flink/streaming/api/environment/ExecutionCheckpointingOptions.java:169]: I see, thanks.

**Review by rkhachatryan — APPROVED** [review]
